geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMemberData.java

```diff
@@ -507,8 +507,8 @@ public void setVmKind(int vmKind) {
 
 
   @Override
-  public void setVersion(Version v) {
-    setVersionOrdinal(v.ordinal());
+  public void setVersion(VersionOrdinal versionOrdinal) {
+    setVersionOrdinal(versionOrdinal.ordinal());
```

Comment:
Let's set `versionOrdinal` (object reference) directly here and eliminate `MemberData.setVersionOrdinal(short)` entirely. The method shouldn't exist at all. We should not be passing a `short` ordinal to a `MemberData`.

There are three places `MemberData.setVersionOrdinal(short)` is called:
1. here: just set the field directly
2. `readEssentialData()`: replace `setVersionOrdinal(VersioningIO.readOrdinal(in));` with `setVersion(Versioning. getVersionOrdinal(VersioningIO.readOrdinal(in)))`
3. `GMSMemberDataVersionJUnitTest.testSetVersionOrdinal()`: delete this test method